hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/deployment/JetClassLoader.java

```diff
@@ -80,9 +80,37 @@ public JetClassLoader(
                     + " or start all members with it on classpath");
         }
         byte[] classBytes = uncheckCall(() -> IOUtil.toByteArray(classBytesStream));
+        definePackage(name);
         return defineClass(name, classBytes, 0, classBytes.length);
     }
 
+    /**
+     * Defines the package if it is not already defined for the given class
+     * name.
+     *
+     * @param className the class name
+     */
+    private void definePackage(String className) {
+        if (isEmpty(className)) {
+            return;
+        }
+        int lastDotIndex = className.lastIndexOf('.');
+        if (lastDotIndex == -1) {
+            return;
+        }
+        String packageName = className.substring(0, lastDotIndex);
+        if (getPackage(packageName) != null) {
```

Comment:
This method was deprecated in java9. I've read a bit why, but am not sure how to fix it in java8. But it's only an issue if multiple class loaders define the same packages, but I'd say it's not common. The purpose of defining the package is to make package annotations available, and if one uses package annotations, he doesn't use a package name that's likely to be used in some other library...

So consider this comment just a thinking aloud, we can live with this code for a while :)

Reply:
Maybe we can rather rely on ignoring the `IArgExc`. Or re-verify there just, as `URLClassLoader` does.